File: app/iosApp/iosApp/UI/Components/ToastView.swift
import SwiftUI
import SharedLogic

struct ToastView: View {
    let state: ToastState
    let designSystem: DesignSystem
    let onDismiss: () -> Void

    var body: some View {
        VStack {
            HStack {
                Text(getMessage())
                    .font(.subheadline)
                    .foregroundColor(getTextColor())
                Spacer()
            }
            .padding(.horizontal, 20)
            .padding(.vertical, 12)
            .background(getBgColor())
            .cornerRadius(12)
            .padding(.horizontal, 16)
            .shadow(radius: 4)
            
            Spacer()
        }
        .padding(.top, 4)
        .transition(.move(edge: .top).combined(with: .opacity))
        .onAppear {
            DispatchQueue.main.asyncAfter(deadline: .now() + 3) {
                onDismiss()
            }
        }
    }

    private func getMessage() -> String {
        if let token = state.messageToken {
            return designSystem.string(token: token)
        }
        return state.messageRaw ?? ""
    }

    private func getBgColor() -> Color {
        switch state.type {
        case .success: return IosTheme.color(.truth, from: designSystem)
        case .error: return IosTheme.color(.stress, from: designSystem)
        case .warning: return IosTheme.color(.primary, from: designSystem)
        default: return .gray
        }
    }

    private func getTextColor() -> Color {
        switch state.type {
        case .success, .error: return IosTheme.color(.textInverted, from: designSystem)
        case .warning: return IosTheme.color(.textPrimary, from: designSystem)
        default: return .white
        }
    }
}

File: app/androidApp/src/main/kotlin/application/liedetector/ui/components/state/ToastView.kt
package application.liedetector.ui.components.state

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.unit.dp
import application.liedetector.theme.utils.composeColor
import application.liedetector.uicore.state.*
import application.liedetector.uicore.theme.*
import kotlinx.coroutines.delay
import kotlin.time.Duration.Companion.milliseconds

@Composable
fun ToastView(
    state: ToastState,
    onDismiss: () -> Unit
) {
    val designSystem = LocalDesignSystem.current

    val bgColor = when (state.type) {
        ToastType.SUCCESS -> designSystem.composeColor(ColorToken.TRUTH)
        ToastType.ERROR -> designSystem.composeColor(ColorToken.STRESS)
        ToastType.WARNING -> designSystem.composeColor(ColorToken.PRIMARY)
    }

    val textColor = when (state.type) {
        ToastType.SUCCESS, ToastType.ERROR -> designSystem.composeColor(ColorToken.TEXT_INVERTED)
        ToastType.WARNING -> designSystem.composeColor(ColorToken.TEXT_PRIMARY)
    }

    val message = state.messageToken?.let { designSystem.string(it) } ?: state.messageRaw ?: ""

    LaunchedEffect(state) {
        delay(3000.milliseconds)
        onDismiss()
    }

    Box(
        modifier = Modifier
            .fillMaxSize()
            .padding(designSystem.dimen(DimenToken.SPACING_LARGE).dp),
        contentAlignment = Alignment.TopCenter
    ) {
        Box(
            modifier = Modifier
                .clip(RoundedCornerShape(designSystem.dimen(DimenToken.CORNER_RADIUS).dp))
                .background(bgColor)
                .padding(horizontal = 24.dp, vertical = 12.dp)
        ) {
            Text(
                text = message,
                color = textColor,
                style = MaterialTheme.typography.bodyMedium
            )
        }
    }
}
